pyro/contrib/epidemiology/seir.py

```diff
@@ -136,3 +108,129 @@ def transition_bwd(self, params, prev, curr, t):
         pyro.sample("obs_{}".format(t),
                     dist.ExtendedBinomial(E2I, rho),
                     obs=self.data[t])
+
+
+class OverdispersedSEIRModel(CompartmentalModel):
+    r"""
+    Overdispersed Susceptible-Exposed-Infected-Recovered model.
+
+    To customize this model we recommend forking and editing this class.
+
+    This is a stochastic discrete-time discrete-state model with four
+    compartments: "S" for susceptible, "E" for exposed, "I" for infected,
+    and "R" for recovered individuals (the recovered individuals are
+    implicit: ``R = population - S - E - I``) with transitions
+    ``S -> E -> I -> R``.
+
+    This model accounts for superspreading (overdispersed individual
+    reproductive number) by assuming each infected individual infects
+    BetaBinomial-many susceptible individuals, where the BetaBinomial
+    distribution acts as an overdispersed Binomial distribution, adapting the
+    more standard NegativeBinomial distribution that acts as an overdispersed
+    Poisson distribution [1,2] to the setting of finite populations. To
+    preserve Markov structure, we follow [2] and assume all infections by a
+    single individual occur on the single time step where that individual makes
+    an ``I -> R`` transition.
+
+    **References**
+
+    [1] J. O. Lloyd-Smith, S. J. Schreiber, P. E. Kopp, W. M. Getz (2005)
+        "Superspreading and the effect of individual variation on disease
+        emergence"
+        https://www.nature.com/articles/nature04153.pdf
+    [2] Lucy M. Li, Nicholas C. Grassly, Christophe Fraser (2017)
+        "Quantifying Transmission Heterogeneity Using Both Pathogen Phylogenies
+        and Incidence Time Series"
+        https://academic.oup.com/mbe/article/34/11/2982/3952784
+
+    :param int population: Total ``population = S + E + I + R``.
+    :param float incubation_time: Mean incubation time (duration in state
+        ``E``). Must be greater than 1.
+    :param float recovery_time: Mean recovery time (duration in state
+        ``I``). Must be greater than 1.
+    :param iterable data: Time series of new observed infections, i.e. a
+        Binomial subset of the ``E -> I`` transitions at each time step.
```

Comment:
nit: i guess this is already merged but isn't 'binomial subset' strange language? maybe something like 
"i.e. a noisily observed time series of the E->I transitions at each time step"

Reply:
How about this?
```
:param iterable data: Time series of new observed infections.
    Each time step is Binomial distributed between 0 and the number of ``E -> I`` transitions.
    This allows false negative but no false positives.
```

Reply:
lgtm. note this also occurs below

Reply:
done